realtek: Zyxel GS1900-48 dts fixes

jwh opened the pull request:
* Use SDS for phy 48/49
 * Use correct link/phy settings for SFP ports
 * Remove read-only flag from u-boot env so fw_setenv actually works

No functional changes

jwh committed:
realtek: Zyxel GS1900-48 dts fixes

 * Use SDS for phy 48/49
 * Use correct link/phy settings for SFP ports
 * Remove read-only flag from u-boot env so fw_setenv actually works

Signed-off-by: Joe Holden <jwh@zorins.us>
Link: https://github.com/openwrt/openwrt/pull/19596
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

plappermaul reviewed (approved):
perfect as discussed in the issue thread.

Djfe commented:
please sign your commit :)
```
git commit --amend -x
git push -f
```

bevanweiss commented:
am I missing the code change that does the 

- Remove read-only flag from u-boot env so fw_setenv actually works

part?

jwh committed:
uboot-envtools: gs1900-48: set fw_env/sys.config

GS1900-48 is missing from package/boot/uboot-tools/uboot-envtools/files/realtek, so add it to be consistent.

Signed-off-by: Joe Holden <jwh@zorins.us>
Link: https://github.com/openwrt/openwrt/pull/19596
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

jwh commented:
I've tacked on missing fw_env/sys.config also to this PR

plappermaul reviewed (changes requested):
Please add at least one sentence to this commit 😊

howels commented:
> check with @howels or @andyboeh . IIRC on of them has those kind of switches.
> 

I've got a unit to test on, will any 1G SFP work? I have a 1Gbase-T SFP which I use to manage the 10G SFP+ switches but no fiber.

jwh commented:
> > check with @howels or @andyboeh . IIRC on of them has those kind of switches.
> 
> I've got a unit to test on, will any 1G SFP work? I have a 1Gbase-T SFP which I use to manage the 10G SFP+ switches but no fiber.

It's more the question of if u-boot-env[2] should be read-only or not, there isn't any change re SFPs at the moment

Djfe commented:
I assume the stock rom had a dual partition feature and was able to boot between A/B partitions? jffs and jffs2 and env and env2 remain of that while firmware merges both, is that right?
If that is the case you can allow env2 to be changed and add it to the ubootenv tool but it is irrelevant to OpenWrt's boot since the second slot is never going to be used, right?
But it doesn't matter too much. Just add the second environment like you did here and be done with it for now :)

jffs and jffs2 could also be merged at some point later in time if you need more space (2MiB instead of 1MiB) - unless the bootloader takes issue with that.

howels commented:
> I assume the stock rom had a dual partition feature and was able to boot between A/B partitions? jffs and jffs2 and env and env2 remain of that while firmware merges both, is that right? If that is the case you can allow env2 to be changed and add it to the ubootenv tool but it is irrelevant to OpenWrt's boot since the second slot is never going to be used, right? But it doesn't matter too much. Just add the second environment like you did here and be done with it for now :)
> 
> jffs and jffs2 could also be merged at some point later in time if you need more space (2MiB instead of 1MiB) - unless the bootloader takes issue with that.

Correct, the stock firmware had 2 partitions but as this was only a 16MB flash device that proved highly restrictive and with 24.10 the partition layout changed to use the entire 16MB for OpenWRT, prior to that we were trying to fit everything into less than 8MB which would have limited the longevity of these.

jwh commented:
> > I assume the stock rom had a dual partition feature and was able to boot between A/B partitions? jffs and jffs2 and env and env2 remain of that while firmware merges both, is that right? If that is the case you can allow env2 to be changed and add it to the ubootenv tool but it is irrelevant to OpenWrt's boot since the second slot is never going to be used, right? But it doesn't matter too much. Just add the second environment like you did here and be done with it for now :)
> > jffs and jffs2 could also be merged at some point later in time if you need more space (2MiB instead of 1MiB) - unless the bootloader takes issue with that.
> 
> Correct, the stock firmware had 2 partitions but as this was only a 16MB flash device that proved highly restrictive and with 24.10 the partition layout changed to use the entire 16MB for OpenWRT. Prior to 24.10 we were trying to fit everything into less than 8MB which would have limited the longevity of these.

I've been running with single partition (combining all available space including the extra 1MB) since well before the change went into main as I immediately hit the ~7MB limit, there have been no issues with bootloader or anything on any of my devices.
